### ISSUE TITLE
CT: [neurites] Missing matrix entries

### ISSUE THREAD

**Issue by drdiek** [Hippocampome-Org/php#484]
In the Dendritic and Axonal Lengths matrix and the Somatic Distances of Dendrites and Axons matrix, there are entries for particular neuron types that are missing, despite the relevant values appearing in the original spreadsheets and the database. Some of the neuron types with missing entries are the following:
DG Basket CCK+ (unique ID 1036)
CA3 Horizontal AA (unique ID 2047)
CA3 Basket CCK+ (unique ID 2044)
CA3 Interneuron Specific Oriens (unique ID 2021)
Mossy Fiber-Associated ORDEN (unique ID 2035)

**Comment by nmsutton** [comment]
@drdiek I have added an update that appears to correct this. I found that it appears I was generating the matrix based on a list of neuron class names that I would guess I retrieved from an original hco table’s list of short names for them instead of neurite_quantified’s list of short names. I corrected the list and the missing values are now showing up. Please close this issue if it looks fine.

**Comment by drdiek** [comment]
@nmsutton I am a little confused here. It is nice that you seem to have gotten things working, but I am concerned about how you may have gone about solving the problem. I do not understand why you are keying off of names of any sort. You should be keying off of the 4 digit Unique ID's that were created to concretely identify each neuron type. The names can fluctuate.

**Comment by nmsutton** [comment]
@drdiek That is a good point that I could use neuron ids. I have now updated the matrix creation of all 5 matrices to use that method. It seems to have produced the same results as the other method, but just for being safe about it, it is good to use the method you described.

I am guessing I originally used the names because neurite_quantified does not provide parcel ids and I needed to map names to ids for that, even though it does provide neuron ids. I may have not wanted to write a query with neuron_ids and parcel_names and instead just used all names to keep it simple, but as you point out it can be safer to use neuron_ids.
